processing/src/main/java/io/druid/segment/data/CompressionFactory.java

```diff
@@ -0,0 +1,275 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment.data;
+
+import com.google.common.base.Supplier;
+import com.google.common.collect.Maps;
+import org.apache.commons.lang.ArrayUtils;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.Map;
+
+/**
+ * Compression of metrics is done by using a combination of {@link io.druid.segment.data.CompressedObjectStrategy.CompressionStrategy}
+ * and Encoding(such as {@link LongEncoding} for type Long). CompressionStrategy is unaware of the data type
+ * and is based on byte operations. It must compress and decompress in block of bytes. Encoding refers to compression
+ * method relies on data format, so a different Encoding exist for each data type.
+ * <p>
+ *
+ * Compression Storage Format
+ * Byte 1 : version (currently 0x02)
+ * Byte 2 - 5 : number of values
+ * Byte 6 - 9 : size per block (even if block format isn't used, this is needed for backward compatibility)
+ * Byte 10 : compression strategy (contains a flag if there's an encoding byte, see below for how the flag is defined)
+ * Byte 11(optional) : encoding type
+ * <p>
+ * Encoding specific header (described below)
+ * <p>
+ * Block related header (if block compression is used, described in GenericIndexed)
+ * <p>
+ * Values
+ */
+public abstract class CompressionFactory
```

Comment:
I think there are no instances of this class, right? If not, drop the "abstract", there's no need for it. You can also signify that this is really just a bag of methods by having a private constructor, which disables the ability for anyone to instantiate it.

example:

```java
private CompressionFactory() {
  // No instantiation
}
```